great_expectations/experimental/datasources/interfaces.py

```diff
@@ -168,6 +254,27 @@ def get_asset(self, asset_name: str) -> DataAssetType:
                 f"'{asset_name}' not found. Available assets are {list(self.assets.keys())}"
             ) from exc
 
+    def add_asset(self, asset: DataAssetType) -> DataAssetType:
+        """Adds an asset to a datasource
+
+        Args:
+            asset: The DataAsset to be added to this datasource.
+        """
+        # The setter for datasource is non-functional so we access _datasource directly.
+        # See the comment in DataAsset for more information.
+        asset._datasource = self
+        self.assets[asset.name] = asset
+        return asset
+
+    # Abstract Methods
+    def execution_engine_type(self) -> Type[ExecutionEngine]:
```

Comment:
Does this get implemented as a property?

It would also seem like `execution_engine_type` could just be a field on this pydantic model and this method could be a [validator](https://docs.pydantic.dev/usage/validators/).

Reply:
Oh, this is a hack to avoid circular dependencies. I don't recall the specifics but we run into this issue because of the relationship between `ExecutionEngine`'s and the `Batch*` types from outside of `Experimental`. When we try to import both, we run into issues. By allowing this to be a function we can delay importing `ExecutionEngine`. That is the implementations look like:

```
    def execution_engine_type(self) -> Type[ExecutionEngine]:
        """Return the PandasExecutionEngine unless the override is set"""
        from great_expectations.execution_engine.pandas_execution_engine import (
            PandasExecutionEngine,
        )

        return PandasExecutionEngine
```

If you think this is better as a property, I could change it to that instead of a function. I could do that here but would prefer to do it in an immediate followup PR to keep the scope of this one from getting bigger.

Reply:
A property would be nice, but that might even go away eventually if we are able to handle the circular imports in the future.

Reply:
👍 , I'll do that as a followup.